### PR TITLE
markdown support

### PR DESCRIPTION
**Pull request by paulirish**
There are lots of backtick blocks these days. And there's even one list:

Before and after:

image

Right now i'm passing the markdown in as an attribute. Is it possible to just have it as the literal innerHTML and then transform it? It didn't seem straightforward, but I'm a polymer noob. 
Given that this is fairly equivalent to what we had, I'm pretty comfortable with this anyway.

### CODE REVIEW

File: elements/cr-markdownish/cr-markdownish.html [hunk 1]
```diff
@@ -0,0 +1,136 @@
+<link rel="import" href="../../bower_components/polymer/polymer.html">
```
Comment (kdzwinel) [inline elements/cr-markdownish/cr-markdownish.html]: oh wow, you wrote your own thing! I wonder why you haven't decided to go with [marked-element](https://www.webcomponents.org/element/PolymerElements/marked-element) or a custom wrapper element for [marked](https://github.com/markedjs/marked) (it's 6kb gziped) ?

File: elements/cr-markdownish/cr-markdownish.html [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        if (!this.markdown) return;
+
+        // skip the work if we don't have markdown in it.
+        const hasMarkdown = /(\n-)|`/.test(this.markdown);
+        if (!hasMarkdown) {
+          Polymer.dom(this).innerHTML = this.markdown;
+          return;
+        }
+
+        let text = this._escapeHtml(this.markdown);
```
Comment (kdzwinel) [inline elements/cr-markdownish/cr-markdownish.html]: nit: can be const

File: elements/cr-markdownish/cr-markdownish.html [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+          }
+          if (line.startsWith('-')) {
+            if (!inList) html.push('<ul>');
+            inList = true;
+            html.push('  <li>' + line.replace(/^- /, ''));
+            continue;
+          }
+          html.push(line);
+        }
+        return html.join('\n');
```
Comment (kdzwinel) [inline elements/cr-markdownish/cr-markdownish.html]: if text ends with a list and there is no blank line at the end this code will miss closing `</ul>`. We should check `inList` after the loop to make sure it's added.

File: elements/cr-markdownish/cr-markdownish.html [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        let inList = false;
+        let html = [];
+
+        for (const line of text.split(/\n/)) {
+          if (inList && line === '') {
+            inList = false;
+            html.push('</ul>');
+            continue;
+          }
+          if (line.startsWith('-')) {
```
Comment (kdzwinel) [inline elements/cr-markdownish/cr-markdownish.html]: not sure if we support IE11 ATM (probably not?), but `startsWith` is not available there

File: elements/cr-markdownish/cr-markdownish.html (hunk 1 above)
Comment (paulirish) [reply elements/cr-markdownish/cr-markdownish.html]: i started with the convertMarkdownLinkSnippets() fn from lighthouse, and figured doing lists wouldn't be too bad either. In the end it probably woulda been workwhile to use marked instead but hey 200bytes vs 6000 seems worth it. :)

File: elements/cr-markdownish/cr-markdownish.html (hunk 3 above)
Comment (paulirish) [reply elements/cr-markdownish/cr-markdownish.html]: good call.

File: elements/cr-markdownish/cr-markdownish.html (hunk 4 above)
Comment (paulirish) [reply elements/cr-markdownish/cr-markdownish.html]: thx for the heads up. i figure its fine if we're Edge and up.

File: elements/cr-markdownish/cr-markdownish.html [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+      attached: function () {
+        this.render();
+      },
+      render: function () {
+        if (!this.markdown) return;
+
+        // skip the work if we don't have markdown in it.
+        const hasMarkdown = /(\n-)|`/.test(this.markdown);
+        if (!hasMarkdown) {
+          Polymer.dom(this).innerHTML = this.markdown;
```
Comment (TimvdLippe) [inline elements/cr-markdownish/cr-markdownish.html]: Should not be necessary with Polymer 2.

File: elements/cr-markdownish/cr-markdownish.html [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+          return;
+        }
+
+        let text = this._escapeHtml(this.markdown);
+        let html = this._convertMarkdownLists(text);
+        html = this._convertMarkdownCodeBlocks(html);
+        Polymer.dom(this).innerHTML = html;
```
Comment (TimvdLippe) [inline elements/cr-markdownish/cr-markdownish.html]: Same here

**Comment by paulirish** [comment]
> Ping @paulirish Do you still want to get this in?

Yup! feedback addressed.

**Pull request merged**